src/rebound/rebound.ts

```diff
@@ -19,38 +19,53 @@ export class Rebound {
 
   /**
    * An object used to keep track of the proper window
-   * @property {Window} reciever
+   * @protected {Window} reciever
    * @private
    */
-  private _reciever: Window;
+  protected _reciever: Window;
 
   /**
    * A boolean that tells you if you are inside the iframe or not
-   * @property {Boolean} isChild
+   * @protected {Boolean} isChild
    * @private
    */
-  private _isChild: boolean;
+  protected _isChild: boolean;
 
   /**
    * A string that contains a random id value used to tell rebound that the
    * events are still from rebound
-   * @property {String} events
-   * @private
+   * @private {String} events
+   * @protected
    */
-  private _randId: string;
+  protected _randId: string;
+
+  protected _isHammerClient: boolean;
 
   /**
    * An object used to keep track of the client instance
    * @property {Object} client
-   * @private
+   * @protected
    */
-  private _client: ClientType;
+  protected _client: ClientType;
 
   constructor() {
     this._isChild = !window.frames.length;
     if (this._isChild) {
       this._randId = 'Rebound_' + (Math.random()).toString();
       this._reciever = parent;
+    }
+  }
+
+  /**
+   * This method creates a connection with the host and starts the message listener
+   * this method should be overridden in case the connect message that is recieved
+   * back is in a different format other than an object. In cases where HammerSpace
+   * is used with a host that is not HammerSpace.
+   * @protected
+   * @method connect
+   */
+  protected _connect() {
```

Comment:
can you add tests for this please and thank you 😛

Reply:
Since we changed the constructor parameter the test cases need to be adjusted, will adjust them along with the unit tests for `_connect`